### DISCUSSION TITLE
Decrypt data on pre-established date

### DISCUSSION THREAD

**Discussion by PetersonFonseca** [filecoin-project/community#514] [Developers]
Hello everyone, I would like to find a way to decrypt data only on a pre-defined date. What would be the best way to do this? I looked for solutions in solana and ethereum but I still haven't found anything....
For example: I send a previously encrypted message, photo or video to Filecoin and on a certain date the file is automatically decrypted.
I know how to easily develop this on web2. My main difficulty on web3 is knowing where to save the private key without exposing it.
Does Filecoin itself have any tools to help with this?

**Comment by nikkolasg** [comment]
Hi @PetersonFonseca,

Filecoin uses [drand](https://drand.love) as its source of randomness and it turns out drand will soon release a new feature that will enable to do what you wish, the "timed encryption" feature. More information here https://drand.love/blog/2022/02/21/multi-frequency-support-and-timelock-encryption-capabilities/
So, _right now_ not on Filecoin, but soon enough hopefully (there is FIP for integrating this new feature once it is in). BUT you can nevertheless use this feature as soon as it is on drand mainnet, or even drand testnet.  I would encourage you to write at leagueofentropy [ at ] googlegroups.com and explain your project so we can better help you.

**Reply by PetersonFonseca** [reply]
Thanks @nikkolasg . The information was very helpful.
